DirectProgramming/DPC++FPGA/PlatformTests/board_test/src/board_test.hpp

```diff
@@ -0,0 +1,1852 @@
+          // Initialize to 0
+          unsigned int pipe_value = 0;
+          // Blocking read from pipe
+          pipe_value = SendertoReceiverPipe::read();
+          auto gid = it.get_global_id(0);
+          mem[gid] = pipe_value;
+        });
+  });
+
+  // **** Wait for sender and receiver kernels to finish **** //
+
+  std::cout << "  ... Waiting for sender\n";
+  e_send.wait();
+  std::cout << "Sender sent the token to receiver\n";
+  std::cout << "  ... Waiting for receiver\n";
+  e_receive.wait();
+
+  // Read back data written by pipe to device memory
+  host_accessor h_buf_access{dev_buf};
+  if (h_buf_access[0] != kTestValue) {
+    std::cout << "Kernel Launch Test failed, incorrrect value read back from "
+              << "pipe between sender and receiver kernel:\n"
+              << "Value written to pipe : " << kTestValue
+              << " Value read back : " << h_buf_access[0] << "\n";
+    return 1;
+  }
+
+  return 0;
+}
+
+////////////////////////////////////
+// **** KernelMemRW function **** //
+////////////////////////////////////
+
+// Inputs:
+// queue &q - queue to submit operation
+// Returns:
+// 0 is test passes, 1 if device memory allocation is 0 or if test fails
+
+// The function does the following tasks:
+// 1. Gets maximum allocation size for device global memory
+// 2. Calculates number of unsigned type elements that can be written to device
+// and allocates device buffer to span this size
+// 3. Allocates host memory for input & output to & from device global memory
+// 4. Write initial data to entire device global memory
+// 5. Launches kernel to modify the values written to device global memory
+// 6. Reads the modified data from device global memory into host memory
+// 7. Verifies data matches expected value
+
+// Following additional function is used in this test:
+// This is defined in this file and declared in corresponding header
+// (kernel_mem_rw.hpp) More details can be found with the corresponsing
+// definition
+// 1. void InitializeVector(unsigned *vector, size_t size, size_t offset)
+
+int ShimMetrics::KernelMemRW(queue &q) {
+  // Test fails if max alloc size is 0
+  if (max_alloc_size == 0) {
+    std::cout << "Maximum global memory allocation supported by Sycl device is "
+              << "0! Cannot run kernel-to-memory read wite test\n\n";
+    return 1;
+  }
+
+  std::cout << "Maximum device global memory allocation size is "
+            << max_alloc_size << " bytes \n";
+
+  // Number of integer type vectors supported on the device
+  size_t max_dev_vectors = max_alloc_size / sizeof(unsigned);
+
+  // **** Host memory Allocation **** //
+
+  // Allocate host vectors
+  // ND Range kernel uses max_dev_vectors to calculate the global range...
+  // ...SYCL ID queries are expected to fit within MAX_INT,...
+  // ...limit the range to prevent trunating data and errors
+  size_t num_host_vectors = (max_dev_vectors > INT_MAX) ? kGB : max_dev_vectors;
```

Comment:
You should use std::numeric_limits to get maximum for types, rather than INT_MAX.
https://en.cppreference.com/w/cpp/types/numeric_limits